packages/neon/neon/lib/src/blocs/apps.dart

```diff
@@ -119,24 +117,26 @@ class AppsBloc extends InteractiveBloc implements AppsBlocEvents, AppsBlocStates
       return;
     }
 
-    final appIds = {
-      'core',
-      ...apps.requireData.map((final a) => a.id),
-    };
+    final notSupported = <(String, String?)>[];
 
-    final notSupported = <(String, Object?)>[];
+    try {
+      final (coreSupported, coreMinimumVersion) = _account.client.core.isSupported(capabilities.requireData);
+      if (!coreSupported) {
+        notSupported.add(('core', coreMinimumVersion.toString()));
+      }
+    } catch (e, s) {
+      debugPrint(e.toString());
+      debugPrint(s.toString());
+    }
 
-    for (final id in appIds) {
+    for (final app in apps.requireData) {
       try {
-        final (supported, minVersion) = switch (id) {
-          'core' => _account.client.core.isSupported(capabilities.requireData),
-          AppIDs.news => await _account.client.news.isSupported(),
-          AppIDs.notes => _account.client.notes.isSupported(capabilities.requireData),
-          _ => (true, null),
-        };
-
-        if (!supported) {
-          notSupported.add((id, minVersion));
+        final support = await app.isSupported(_account, capabilities.requireData);
+        if (support == null) {
+          continue;
+        }
+        if (!support.$1) {
```

Comment:
I'm not a fan of the `.$n` syntax for records.

I'd rather have variable names that I can understand while reading the code

Reply:
Agreed, but then we would have to introduce a class for this (I would be fine with that).

Reply:
I removed this from the PR and will open a second one since the changes are not directly related to this PR anymore.